pkgs/unified_analytics/lib/src/session.dart

```diff
@@ -40,30 +39,27 @@ class Session {
   ///
   /// Note, the file will always be updated when calling this method
   /// because the last ping variable will always need to be persisted.
-  int getSessionId() {
+  int? getSessionId() {
     _refreshSessionData();
     final now = clock.now();
 
     // Convert the epoch time from the last ping into datetime and check if we
     // are within the kSessionDurationMinutes.
-    final lastPingDateTime = DateTime.fromMillisecondsSinceEpoch(_lastPing);
+    final lastPingDateTime = sessionFile.lastModifiedSync();
     if (now.difference(lastPingDateTime).inMinutes > kSessionDurationMinutes) {
-      // In this case, we will need to change both the session id
-      // and the last ping value
+      // Update the session file with the latest session id
       _sessionId = now.millisecondsSinceEpoch;
+      sessionFile.writeAsStringSync('{"session_id": $_sessionId}');
     }
 
-    // Update the last ping to reflect session activity continuing
-    _lastPing = now.millisecondsSinceEpoch;
-
-    // Rewrite the session object back to the file to persist
-    // for future events
-    sessionFile.writeAsStringSync(toJson());
+    // Update the last modified timestamp with the current timestamp so that
+    // we can use it for the next _lastPing calculation
+    sessionFile.setLastModifiedSync(now);
```

Comment:
this can be moved to an else block on line 53--we don't need to set the lastmodified if we just wrote to it from the if block